Better /wiki - Fixed bugs + added official wiki support

Obsidianninja11 opened the pull request.

Obsidianninja11 committed:
Better /wiki - Fixed bugs + added official wiki support

lineargraph commented on src/main/java/at/hannibal2/skyhanni/features/commands/WikiManager.kt:
```diff
@@ -35,21 +29,23 @@ class WikiManager {
         if (message == "/wiki") {
-            LorenzUtils.chat("Opening the Fandom Wiki..")
-            OSUtils.openBrowser("${urlPrefix}Hypixel_SkyBlock_Wiki")
+            LorenzUtils.clickableLinkChat("Click to open the wiki!", urlPrefix,config.autoOpenWiki,"Open the wiki!")
```
Maybe run your auto formatter over this PR: 
```suggestion
            LorenzUtils.clickableLinkChat("Click to open the wiki!", urlPrefix, config.autoOpenWiki, "Open the wiki!")
```

Obsidianninja11 committed:
Better code formatting

Obsidianninja11 committed:
Merge branch 'beta' into better-wiki

lineargraph commented on src/main/java/at/hannibal2/skyhanni/features/commands/WikiManager.kt:
```diff
@@ -35,21 +29,25 @@ class WikiManager {
-                wikiTheItem(itemInHand)
+                wikiTheItem(itemInHand, config.autoOpenWiki)
             } else {
                 val search = event.message.split("/wiki ").last()
```
Why use split here? Even removePrefix would be better (but still flawed, since that one isn't case insensitive). Also *please* urlEncode the search.

```suggestion
                val search = event.message.drop("/wiki ".length)
```

Obsidianninja11 replied:
I didn't make this line.
I think @ErymanthusRayDeeUx did

lineargraph replied:
I didnt say you did it. Either way a semantically wrong thing in your pull request, so i will point it out.

lineargraph commented on src/main/java/at/hannibal2/skyhanni/features/commands/WikiManager.kt:
```diff
@@ -61,17 +59,22 @@ class WikiManager {
-        val wikiUrlSearch = if (internalName != "NONE") "$urlSearchPrefix$internalName&scope=internal"
-        else "$urlSearchPrefix${itemDisplayName.removeColor()}&scope=internal"
-        OSUtils.openBrowser(wikiUrlSearch.replace(' ', '+'))
+        val wikiUrlSearch = if (internalName != "NONE") "$urlSearchPrefix$internalName"
```
Please url encode the internal name

Obsidianninja11 replied:
Is the url encode the `.replace(' ', '+')`?
This is already in the search

lineargraph replied:
See: https://docs.oracle.com/javase/8/docs/api/java/net/URLEncoder.html (or https://url.spec.whatwg.org/#percent-encoded-bytes)

lineargraph commented on src/main/java/at/hannibal2/skyhanni/features/misc/BetterWikiFromMenus.kt:
```diff
@@ -56,22 +61,24 @@ class FandomWikiFromMenus {
-            LorenzUtils.chat("Searching the Fandom Wiki for §a$wikiDisplayName")
-            val wikiUrlCustom = "${WikiManager.urlSearchPrefix}$wikiInternalName&scope=internal"
-            OSUtils.openBrowser(wikiUrlCustom.replace(' ', '+'))
+            val wikiUrlCustom = "${urlSearchPrefix}$wikiInternalName&scope=internal"
```
Please url encode the internal name

hannibal002 reviewed (changes requested):
Clicking on 
![image](https://github.com/hannibal002/SkyHanni/assets/24389977/9b209feb-69a0-40f3-8d99-49521ee2bf69)
with fandom wiki disabled, opens 
https://wiki.hypixel.net/index.php?search=Livid+Collection+7&scope=internal

This is definitely not intended, as the search results are useless in this situation.
Disabling this feature, Hypixel links to 
https://wiki.hypixel.net/Livid#Collection

maybe simply read the chat message from hypixel if fandom wiki is enabled instead?

Obsidianninja11 committed:
Code cleanup + more

Obsidianninja11 committed:
Merge branch 'beta' into better-wiki

Obsidianninja11 commented:
> Clicking on ![image](https://private-user-images.githubusercontent.com/24389977/302274569-9b209feb-69a0-40f3-8d99-49521ee2bf69.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDczNjQ4NDYsIm5iZiI6MTcwNzM2NDU0NiwicGF0aCI6Ii8yNDM4OTk3Ny8zMDIyNzQ1NjktOWIyMDlmZWItNjlhMC00MGYzLThkOTktNDk1MjFlZTJiZjY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA4VDAzNTU0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAzODNkOTAxMjYwYjJlNjRlOTk5ZDc1MTA3OWNlODVlMTFhYTViNWIzNDMwM2FmMjM3NzM1OGU0NjYzMTEwMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZjjLjFjOjm6rdxXDKzsTSPNOWjU03sppOaQZJ_PqVsA) with fandom wiki disabled, opens https://wiki.hypixel.net/index.php?search=Livid+Collection+7&scope=internal
> 
> This is definitely not intended, as the search results are useless in this situation. Disabling this feature, Hypixel links to https://wiki.hypixel.net/Livid#Collection
> 
> maybe simply read the chat message from hypixel if fandom wiki is enabled instead?

It would be complicated to do anything more than this. I can't just hide hypixel's msg and get the same link, because if I don't cancel the event, the inventory will close. The best way to fix this would be to use the repo, but that would be complicated.
I just added a seperate toggle for this instead in this pr.

Obsidianninja11 committed:
Removed debugging code + fixes

Obsidianninja11 commented on src/main/java/at/hannibal2/skyhanni/features/commands/WikiManager.kt:
```diff
@@ -9,21 +9,20 @@ import at.hannibal2.skyhanni.utils.ItemUtils.nameWithEnchantment
+    private const val OFFICIALURLPREFIX = "https://wiki.hypixel.net/"
+    private const val OFFICIALSEARCHPREFIX = "index.php?search="
+    private const val FANDOMURLPREFIX = "https://hypixel-skyblock.fandom.com/wiki/"
+    private const val FANDOMSEARCHPREFIX = "Special:Search?query="
```
Also should these be consts and be all caps?
Intellij told me to do this

hannibal002 replied:
Constants should be upper case, this is correct. Just please add an underline character between words: THIS_WAY

Obsidianninja11 commented:
Oh yeah in code cleanup + more, I added commands for each wiki that work even without the feature disabled, and I changed the look of the msgs to be more in line with the vanilla /wiki commands

Obsidianninja11 committed:
Formatting fixes
